build/Common.props

```diff
@@ -61,4 +61,22 @@
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>-->
   </ItemGroup>
+
+  <ItemGroup Condition="'$(IncludeSharedInstrumentationSource)'=='true'">
```

Comment:
could we split this into 2 conditions:
1.  IncludeDiagnosticSourceSubscriptionHelpers - require for those instrumentations which are dealing with DS based instrumentation. HttpClient/AspNet etc. This contains code specific to subscribing to DS.
2. IncludeInstrumentationHelpers - these are general Activity related helpers and are required for all instrumentations. eg; Redis which requires this, but does not require DS helpers.

Reply:
Yeah, I can do that. Seems unnecessary to me though. They are just extra source files, they aren't hurting anything. Just making it so that libs will need to manage 2 options instead of one.

Reply:
Done

Reply:
Thanks! I asked for it make it very clear that only some instrumentations require the DS related stuff. There were lot of confusion earlier about the need for the DS.* classes. Now its clear thats its only required, if you are dealing with a library already producing DS events, which is only a subset of libraries.
(will be updated the now incorrect section https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace/extending-the-sdk#writing-own-instrumentation-library here)

Reply:
Ok, are the changes I made ok?